LanguageFeatures/Augmentation-libraries/augmenting_functions_A06_t02.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - An augmenting declaration uses augmented when the original declaration has
+///   no concrete implementation. Note that all external declarations are
+///   assumed to have an implementation provided by another external source, and
+///   they will throw a runtime exception when called if not.
+///
+/// @description Checks that it is not an error if an augmenting
+/// declaration uses `augmented` when the original declaration has no concrete
+/// implementation but is an `external` declaration
```

Comment:
I think it's fair to say that an external declaration is assumed to have an implementation. It's written in some other language, but it is assumed to exist (so, for example, it's ok to call an external instance member `foo` in a superinvocation like `super.foo()`).

It is a separate matter that the implementation can be something like `throw "Not bound to an implementation"` in the case where no implementation is provided in any other way.

Perhaps we should just talk about the fact that the declaration is external, and avoid saying anything about whether or not it has an implementation?